### PR TITLE
Add additional indexes, tags, streams, update docs, tests, examples

### PR DESCRIPTION
**Pull request by dannyleesmith**
##### Corresponding Issue(s) or trello card(s):

N/A

##### Summary of change(s):

- add streams
- add LSI
- add GSI
- add tags
- new `terraform-docs` appropriate block at top of main.tf
- new tests

##### Will the change trigger resource destruction or replacement? If yes, please provide justification:

No

##### Does this update/change involve issues with other external modules? If so, please describe the scenario.

No

##### If input variables or output variables have changed or has been added, have you updated the README?

Yes

##### Do examples need to be updated based on changes?

Done

##### Note to the PR requester about Closing PR's
Please message the person that opened the issue when auto closing it on slack, as well as any other stake holders of deep interest. Only close the issue if you believe that the issue is fully resolved with this PR.

#### This PR may auto close the issue associated with it. If you feel the issue is not resolved please reopen the issue.

### CODE REVIEW

**Comment by dannyleesmith** [comment]
Please note, I intended to do more with this PR but due to limitations on blocks within a resource (lack of conditional inclusion, HCL2/TF0.12 hopefully will resolve) I could not find a pattern that would be readily agreeable. For now I limited this PR to the inclusion of GSI and some other quick wins.

The issue I was facing was with the `ttl` block. You can't set it to `false` as that causes it to want to apply on every plan because the state doesn't remember `false`. You _could_ set it to `true` for every customer with a default attribute of say `Expires` but some would not want that. You could make it a two resource job with counts to check if it's TTL or not then do or not build out that attribute, but at some point when upgrades need to happy and this is solved to consolidate to one resource you would need to perform state operations to move it to the new id.

I also would like to add autoscaling at some point but that will need consideration also.

**Comment by mjcmtb** [comment]
Resource destruction is due to changing encryption default to AWS owned CMK

File: README.md [hunk 1]
```diff
@@ -1,21 +1,76 @@
...
+| local\_secondary\_index\_maps | A list of maps for each [local secondary index (LSI)](https://www.terraform.io/docs/providers/aws/r/dynamodb_table.html#local_secondary_index-1). Please see [examples](./examples) for usage. | list | `<list>` | no |
 | point\_in\_time\_recovery | Enable point in time recovery for the table. | string | `"false"` | no |
-| range\_key | ** Forces new resource ** RangeType PrimaryKey Name. If used, it will need to be defined by type in attributes | string | `""` | no |
+| range\_key | **Forces new resource!** RangeType PrimaryKey Name. If used, it will need to be defined by type in attributes | string | `""` | no |
 | read\_capacity\_units | Provisioned read throughput. Should be between 5 and 10000. Ignored if `enable_pay_per_request` is set to `true`. | string | `"5"` | no |
-| table\_encryption | Server side table encryption at rest. i.e. true | false | string | `"true"` | no |
+| stream\_enabled | Enable the stream setting on the table. | string | `"false"` | no |
```
Comment (mjcmtb) [inline README.md:62]: A bit of a nitpick but would like to see this represented closer to a boolean, so `false` without quotes.

File: tests/default/main.tf [hunk 2]
```diff
@@ -3,23 +3,147 @@ provider "aws" {
   region  = "us-west-2"
 }
 
+locals {
+  tags = {
+    Environment     = "Test"
+    ServiceProvider = "Rackspace"
+    Terraform       = "true"
```
Comment (johnctitus) [inline tests/default/main.tf]: I would remove the `Environment` and `ServiceProvider` tags.  They are already set within the module, and might cause confusion for folks down the road.

File: variables.tf [hunk 3]
```diff
@@ -26,25 +50,37 @@ variable "read_capacity_units" {
   default     = 5
 }
 
-variable "table_encryption" {
-  description = "Server side table encryption at rest. i.e. true | false"
+variable "stream_enabled" {
+  description = "Enable the stream setting on the table."
   type        = "string"
-  default     = true
+  default     = "false"
```
Comment (mjcmtb) [inline variables.tf]: Same statement as previous. Should be more of a bool string: `false` without quotes. Would probably updates references in tests as well.

File: main.tf [hunk 4]
```diff
@@ -1,18 +1,85 @@
...
+}
+
+resource "aws_dynamodb_table" "basic-dynamodb-table" {
+  billing_mode     = "${var.enable_pay_per_request ? "PAY_PER_REQUEST" : "PROVISIONED" }"
+  hash_key         = "${var.hash_key}"
+  name             = "${var.table_name}"
+  range_key        = "${var.range_key}"
+  read_capacity    = "${var.read_capacity_units}"
+  stream_enabled   = "${var.stream_enabled}"
+  stream_view_type = "${var.stream_enabled == "true" ? var.stream_view_type : "" }"
```
Comment (mjcmtb) [inline main.tf]: would like to see this as `"${var.stream_enabled ? var.stream_view_type : "" }"`

**Review by johnctitus — CHANGES_REQUESTED** [review]
One minor ask around the tags in the tests, but overall looks good.

**Review by stevengorrell — DISMISSED** [review]

**Review by mjcmtb — CHANGES_REQUESTED** [review]

File: main.tf (hunk 4 above)
Comment (dannyleesmith) [reply main.tf]: I considered this, @mjcmtb , but before I make the change I would put it to you that `var.stream_enabled` would evaluate to true if you provided anything that did not equate to false. Would you want this behaviour or would you expect someone to provide exactly `true` ?

File: variables.tf (hunk 3 above)
Comment (dannyleesmith) [reply variables.tf]: Fixed this one in next commit

File: README.md (hunk 1 above)
Comment (dannyleesmith) [reply README.md:62]: This is not me, but `terraform-docs`

File: main.tf (hunk 4 above)
Comment (mjcmtb) [reply main.tf]: That would make it consistent with the behavior we have in other modules, so I would stick to my original comment. That's also the behavior in many programming languages.

Comment (dannyleesmith) [reply main.tf]: Added to next commit

**Review by mjcmtb — APPROVED** [review]

**Review by stevengorrell — APPROVED** [review]

**Review by johnctitus — APPROVED** [review]
LGTM with latest changes.

**Pull request merged**